mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/mapping/impl/OutboxPollingSearchMappingImpl.java

```diff
@@ -0,0 +1,149 @@
+		this.tenantIds = tenantIds;
+	}
+
+	@Override
+	public long countAbortedEvents() {
+		checkNoTenant();
+
+		AtomicLong result = new AtomicLong();
+		try ( Session session = sessionFactory.openSession() ) {
+			transactionHelper.inTransaction( (SharedSessionContractImplementor) session, null, s -> {
+				Query<Long> query = session.createQuery( COUNT_EVENTS_WITH_STATUS, Long.class );
+				query.setParameter( "status", OutboxEvent.Status.ABORTED );
+				result.set( query.getSingleResult() );
+			} );
+		}
+		return result.get();
+	}
+
+	@Override
+	public long countAbortedEvents(String tenantId) {
+		checkTenant( tenantId );
+
+		AtomicLong result = new AtomicLong();
+		try ( Session session = sessionFactory.withOptions().tenantIdentifier( tenantId ).openSession() ) {
+			transactionHelper.inTransaction( (SharedSessionContractImplementor) session, null, s -> {
+				Query<Long> query = session.createQuery( COUNT_EVENTS_WITH_STATUS, Long.class );
+				query.setParameter( "status", OutboxEvent.Status.ABORTED );
+				result.set( query.getSingleResult() );
+			} );
+		}
+		return result.get();
+	}
+
+	@Override
+	public int reprocessAbortedEvents() {
+		checkNoTenant();
+
+		AtomicInteger result = new AtomicInteger();
+		try ( Session session = sessionFactory.openSession() ) {
+			transactionHelper.inTransaction( (SharedSessionContractImplementor) session, null, s -> {
+				Query<?> query = session.createQuery( UPDATE_EVENTS_WITH_STATUS );
+				query.setParameter( "status", OutboxEvent.Status.ABORTED );
+				query.setParameter( "newStatus", OutboxEvent.Status.PENDING );
+				result.set( query.executeUpdate() );
+			} );
+		}
+		return result.get();
+	}
+
+	@Override
+	public int reprocessAbortedEvents(String tenantId) {
+		checkTenant( tenantId );
+
+		AtomicInteger result = new AtomicInteger();
+		try ( Session session = sessionFactory.withOptions().tenantIdentifier( tenantId ).openSession() ) {
+			transactionHelper.inTransaction( (SharedSessionContractImplementor) session, null, s -> {
+				Query<?> query = session.createQuery( UPDATE_EVENTS_WITH_STATUS );
+				query.setParameter( "status", OutboxEvent.Status.ABORTED );
+				query.setParameter( "newStatus", OutboxEvent.Status.PENDING );
+				result.set( query.executeUpdate() );
+			} );
+		}
+		return result.get();
+	}
+
+	@Override
+	public int clearAllAbortedEvents() {
+		checkNoTenant();
+
+		AtomicInteger result = new AtomicInteger();
+		try ( Session session = sessionFactory.openSession() ) {
+			transactionHelper.inTransaction( (SharedSessionContractImplementor) session, null, s -> {
+				Query<?> query = session.createQuery( DELETE_EVENTS_WITH_STATUS );
+				query.setParameter( "status", OutboxEvent.Status.ABORTED );
+				result.set( query.executeUpdate() );
+			} );
+		}
+		return result.get();
+	}
+
+	@Override
+	public int clearAllAbortedEvents(String tenantId) {
+		checkTenant( tenantId );
+
+		AtomicInteger result = new AtomicInteger();
```

Comment:
Same comment: please use an overload of `TransactionHelper#inTransaction` instead.